dhall-csv/tasty/Main.hs

```diff
@@ -52,8 +54,15 @@ csvToDhallGolden = do
         [ Silver.goldenVsAction
             (takeBaseName csvFile)
             dhallFile
-            (Data.ByteString.readFile dhallFile)
-            Data.Text.Encoding.decodeUtf8
+            (textToCsv =<< Data.Text.IO.readFile csvFile)
+            (D.pretty . Dhall.CsvToDhall.dhallFromCsv)
         | csvFile <- csvFiles
         , let dhallFile = replaceExtension csvFile ".dhall"
         ]
+
+textToCsv :: Text -> IO [Data.Csv.NamedRecord]
+textToCsv txt =
+    case Dhall.Csv.Util.decodeCsvDefault txt of
+        Left err -> error err
```

Comment:
I typically use `fail` instead of `error` in `IO` code, for the reasons in this post:

https://www.haskellforall.com/2019/12/prefer-to-use-fail-for-io-exceptions.html

```suggestion
        Left err -> fail err
```

Reply:
Thanks!